### PR TITLE
feat(baseline-implementation): Add singleWorker Option

### PR DESCRIPTION
**Pull request by AhmedAbdoOrtiga**
#### Details
Add singleWorker option in the GH Action/ ADO extension side.

##### Motivation
Getting deterministic result for the baselining.

##### Context
singleWorker and maxUrls will be used to check if the result is deterministic or not.

#### Pull request checklist

- [ ] Addresses an existing issue: Fixes #0000
- [x] Added relevant unit test for your changes. (`yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] Ran precheckin (`yarn precheckin`)
- [ ] (after PR created) The `Accessibility Checks (pull_request)` check should fail. All other checks should pass.

### CODE REVIEW

File: packages/ado-extension/src/task-config/ado-task-config.spec.ts [hunk 1]
```diff
@@ -52,14 +52,21 @@ describe(ADOTaskConfig, () => {
...
+        ${'singleWorker'}              | ${true}             | ${true}                                                 | ${() => taskConfig.getSingleWorker()}
     `(
         `input value '$inputValue' returned as '$expectedValue' for '$inputOption' parameter`,
         ({ inputOption, getInputFunc, inputValue, expectedValue }) => {
-            adoTaskMock
-                .setup((am) => am.getInput(inputOption))
-                .returns(() => inputValue as string)
-                .verifiable(Times.once());
-
+            if (typeof inputValue == 'boolean') {
```
Comment (lisli1) [inline packages/ado-extension/src/task-config/ado-task-config.spec.ts:59]: It looks like there's a `.each` test for boolean input values on line 76. We could use that for the `singleWorker` input or (not necessarily in this PR) move that one to this updated test

File: packages/shared/src/scanner/crawl-argument-handler.ts [hunk 2]
```diff
@@ -50,6 +50,7 @@ export class CrawlArgumentHandler {
             discoveryPatterns: discoveryPatternsArg?.split(/\s+/),
             inputUrls: inputUrlsArg?.split(/\s+/),
             url: this.taskConfig.getUrl(),
+            //to-do singleWorker: this.taskConfig.getSingleWorker()
```
Comment (lisli1) [inline packages/shared/src/scanner/crawl-argument-handler.ts:53]: Is this comment intentional?

File: packages/ado-extension/src/task-config/ado-task-config.spec.ts (hunk 1 above)
Comment (AhmedAbdoOrtiga) [reply packages/ado-extension/src/task-config/ado-task-config.spec.ts:59]: I'll remove the .each for booleans and merge them all in one

File: packages/shared/src/scanner/crawl-argument-handler.ts (hunk 2 above)
Comment (AhmedAbdoOrtiga) [reply packages/shared/src/scanner/crawl-argument-handler.ts:53]: Yeah

**Review by lisli1 — APPROVED** [review]

**Pull request merged**